sirepo/jupyterhub.py

```diff
@@ -56,16 +55,10 @@ async def refresh_user(self, user, handler=None):
 
     def _check_permissions(self, handler):
         r = requests.post(
-            _cfg.sirepo_uri + sirepo.simulation_db.SCHEMA_COMMON.route.checkAuthJupyterhub,
+            self.sirepo_uri + sirepo.simulation_db.SCHEMA_COMMON.route.checkAuthJupyterhub,
```

Comment:
Seems like `+` is a bit too loose. If there is no `/`, the URL is broken. Is there a guarantee on these routes that they begin with `/`. Also, is there a guarantee there's no params? Seems like sirepo_uri should be the complete uri, and that's configured by rsconf. Then you wouldn't need to have simulation_db afaict.

It looks like most of the import sirepo.* are unnecessary now. This would fully decouple sirepo/jupyterhub.py.

Reply:
Added a POSIT for no params and fixed imports 8f4e58209

SCHEMA_COMMON.route is [guaranteed to start with a slash](https://github.com/radiasoft/sirepo/blob/78abea726afdd196a064a36d87dacf5c52bc0b0a/sirepo/uri_router.py#L316). 

I don't like having the complete uri configured by rsconf. Right now if you search for the word `checkAuthJupyterhub` you can see it in the route definition in schema_common, the `def api_checkAuthJupyterhub, and here. If we move to rsconf I think we lose context about how this all works. Someone would have to know to check rsconf (maybe they should since all cfg things get configured by rsconf but I don't think that is established knowledge on the team) . If someone were to change the route (or add params to the route) then they'd also have to fix rsconf. If it is just here then they at least stand a chance of fixing all of the places the route is used.

Reply:
Fair enough.